docs/DesignDocs/TEECallsRefactoring.md

```diff
@@ -0,0 +1,596 @@
+Generalize the Calling Mechanisms of TEE Functions
```

Comment:
Does this title still apply? Also the PR title seems a bit broader than the new scope of this proposal.

Reply:
Updated title.